### PR TITLE
Add options to skip GitHub Branch Source notifications

### PR DESCRIPTION
**Pull request by XiongKezhi**
image

### CODE REVIEW

File: src/main/resources/io/jenkins/plugins/checks/github/status/GitHubSCMSourceStatusChecksTrait/config.jelly [hunk 1]
```diff
@@ -9,4 +9,8 @@
     <f:textbox default="Jenkins"/>
   </f:entry>
 
+  <f:entry title="Skip GitHub Branch Source notifications" field="skipNotifications">
+    <f:checkbox default="false"/>
```
Comment (timja) [inline src/main/resources/io/jenkins/plugins/checks/github/status/GitHubSCMSourceStatusChecksTrait/config.jelly:13]: shouldn't be needed
```suggestion
    <f:checkbox />
```

**Review by timja — APPROVED** [review]

**Comment by medianick** [comment]
Is this meant to obviate the use of the [Skip Notifications Trait plugin](https://plugins.jenkins.io/skip-notifications-trait)?

**Comment by timja** [comment]
> Is this meant to obviate the use of the [Skip Notifications Trait plugin](https://plugins.jenkins.io/skip-notifications-trait)?

yes it means you no longer need that plugin if you're using the github checks and want to suppress the default notification

**Comment by medianick** [comment]
Thanks @timja. I have noticed some of the older `continuous-integration/jenkins/pr-merge` statuses sneaking in -- usually as aborted, tied to the next-to-latest build -- in recent days, probably since our upgrade to GitHub Checks 1.0.9 (although I couldn't swear to it). That's despite using the aforementioned "Skip Notifications" trait, but without this new "Skip GitHub Branch Source notifications" being checked. Is there anything about this implementation that would make that more likely to occur? Previously the trait alone was sufficient to stop all old-style statuses from showing up.

**Comment by timja** [comment]
Not that I’m aware of

**Comment by XiongKezhi** [comment]
> Is there anything about this implementation that would make that more likely to occur? Previously the trait alone was sufficient to stop all old-style statuses from showing up.

@medianick thanks for the report, I figured out that if you are using these two plugins at the same time, and you didn't check the `Skip GitHub Branch Source notifications`, we may end up re-triggering the notifications that have been disabled by skip notifications trait plugin. 

https://github.com/jenkinsci/github-checks-plugin/blob/cdd948abbf8617779171bc706e67092179ce9036/src/main/java/io/jenkins/plugins/checks/github/status/GitHubSCMSourceStatusChecksTrait.java#L93-L95

Adding a condition to check if the notification has been disabled can fix that.

**Pull request merged**